dreamerv2/evaluate_reconstructions.py

```diff
@@ -24,113 +30,56 @@
-                          fixed_goal=False,
-                          depot_dropoff_only=True,
-                          should_render=True,
-                          dimensions=TaxiEnv.dimensions_5x5_to_64x64)
-            env = NoiseWrapper(env, sigma=0.01)
-            env = ClipWrapper(env, 0.0, 1.0)
-            env = TransformWrapper(env, lambda x: x - 0.5)
-            env = common.GymWrapper(env)
-            env = common.ResizeImage(env)
-
-            if hasattr(env.act_space['action'], 'n'):
-                env = common.OneHotAction(env)
-            else:
-                env = common.NormalizeAction(env)
-        else:
-            raise NotImplementedError(suite)
-        env = common.TimeLimit(env, config.time_limit)
-        return env
-
     def train_step(trajectory):
-
         trajectory = {
-                k: np.expand_dims(np.array([convert(experience[k]) for experience in [trajectory]]),0) for k in trajectory.keys()}
-
+            k: np.expand_dims(np.array([convert(experience[k]) for experience in [trajectory]]), 0)
+            for k in trajectory.keys()
+        }
         return trajectory
-    
+
     def reconstruction_prediction(agnt, trajectory, key):
-        
         embed = agnt.wm.encoder(trajectory)
-        
-        states, _ = agnt.wm.rssm.observe(embed[:6, :5], trajectory['action'][:6, :5], trajectory['is_first'][:6, :5])
-
+        states, _ = agnt.wm.rssm.observe(embed[:6, :5], trajectory['action'][:6, :5],
+                                         trajectory['is_first'][:6, :5])
         recon = agnt.wm.heads['decoder'](agnt.wm.rssm.get_feat(states))[key].mode()[:6]
-
         init = {k: v[:, -1] for k, v in states.items()}
 
         prior = agnt.wm.rssm.imagine(trajectory['action'][:6, 5:], init)
         openl = agnt.wm.heads['decoder'](agnt.wm.rssm.get_feat(prior))[key].mode()
 
-        model_prediction = tf.concat([recon[:,:5] + 0.5, openl + 0.5], 1)
+        model_prediction = tf.concat([recon[:, :5] + 0.5, openl + 0.5], 1)
         ground_truth = trajectory[key][:6] + 0.5
 
-        error = (model_prediction - ground_truth + 1)/2
+        error = (model_prediction - ground_truth + 1) / 2
 
         image = tf.concat([ground_truth, model_prediction, error], 2)
         B, T, H, W, C = image.shape
         image.transpose((1, 2, 0, 3, 4)).reshape((T, H, B * W, C))
 
         return model_prediction, ground_truth, error
-    
+
     parser = argparse.ArgumentParser()
     parser.add_argument('--model_path', required=True, type=str)
 
-
     args = parser.parse_args()
 
     logdir = pathlib.Path(args.model_path).expanduser()
 
     #check if logdir is valid
-    if not logdir.exists() or not (logdir/'saved_models').exists() or os.listdir((logdir/'saved_models'))==0:
+    if not logdir.exists() or not (logdir / 'saved_models').exists() or os.listdir(
+        (logdir / 'saved_models')) == 0:
         raise RuntimeError(f'Models not found in {logdir} or [...]/saved_models')
```

Comment:
Copy this check over to api_support_functions.py in place of the `NotImplementedError`

Reply:
Done